File: apps/mobile/modules/tripin-amap/android/src/main/java/expo/modules/tripinamap/TripinAmapModule.kt
package expo.modules.tripinamap

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class TripinAmapModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("TripinAmap")

    View(TripinAmapView::class) {
      OnViewDestroys { view: TripinAmapView ->
        view.destroy()
      }

      Prop("markers") { view: TripinAmapView, markers: List<Map<String, Any?>>? ->
        view.setMarkers(markers)
      }

      Prop("polylines") { view: TripinAmapView, polylines: List<Map<String, Any?>>? ->
        view.setPolylines(polylines)
      }
    }
  }
}

File: apps/mobile/modules/tripin-amap/ios/TripinAmapModule.swift
import ExpoModulesCore

public final class TripinAmapModule: Module {
  public func definition() -> ModuleDefinition {
    Name("TripinAmap")

    View(TripinAmapView.self) {
      Prop("markers") { (view: TripinAmapView, markers: [[String: Any]]?) in
        view.markers = markers ?? []
      }

      Prop("polylines") { (view: TripinAmapView, polylines: [[String: Any]]?) in
        view.polylines = polylines ?? []
      }
    }
  }
}
